modules/programs/vscode.nix

```diff
@@ -221,7 +228,22 @@ in {
   };
 
   config = mkIf cfg.enable {
-    home.packages = [ cfg.package ];
+    home.packages = if cfg.extraPackages != [ ] then
+      [
+        (pkgs.symlinkJoin {
+          name =
+            "${lib.getName cfg.package}-wrapped-${lib.getVersion cfg.package}";
+          paths = [ cfg.package ];
+          preferLocalBuild = true;
+          nativeBuildInputs = [ pkgs.makeWrapper ];
+          postBuild = ''
+            wrapProgram $out/bin/code \
```

Comment:
I believe this won't work if `cfg.package` is `vscodium`, since its main executable is called [`codium`](https://github.com/NixOS/nixpkgs/blob/612f97239e2cc474c13c9dafa0df378058c5ad8d/pkgs/applications/editors/vscode/vscodium.nix#L65)


Maybe something like `$out/bin/${cfg.package.meta.mainProgram or (lib.getName cfg.package)}`? Like in [`lib.getExe`](https://github.com/hsjobeki/nixpkgs/blob/09fb8770db06195580ab47b22cf9722107df8ea1/lib/meta.nix#L252-L256)